### PR TITLE
TMA-1275: Skip cloning ADDv2 process into LCM master

### PR DESCRIPTION
**Pull request by vu-pham**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1551/ : SUCCESS in 7s

File: bin/test_projects_cleanup.rb [hunk 1]
```diff
@@ -86,6 +86,7 @@ def clean_up!(client, force, days)
   delete_project_by_title(/LCM spec Client With Conflicting LDM Changes/, projects, days, force)
   delete_project_by_title(/LCM spec master project/, projects, days, force)
   delete_project_by_title(/users brick load test/, projects, days, force)
+  delete_project_by_title(/#transfer_processes and #transfer_schedules test/, client, days, force)
```
Comment (juras-gdc) [inline bin/test_projects_cleanup.rb]: no "client", but "projects"

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1552/ : SUCCESS in 19s

File: bin/test_projects_cleanup.rb (hunk 1 above)
Comment (vu-pham) [reply bin/test_projects_cleanup.rb]: fixed

File: lib/gooddata/models/process.rb [hunk 2]
```diff
@@ -186,14 +186,18 @@ def deploy_component(data, options = { client: GoodData.client, project: GoodDat
...
-                      value: true
-                    }
+        data[:process][:component] = GoodData::Helpers.symbolize_keys(data[:process][:component]).select { |k| %i[name version configLocation config].include? k }
+
+        # a feature flag is required to create ADDv2 components
+        if data[:process][:component][:name] == 'gdc-data-distribution'
+          client.post(
+            "/gdc/projects/#{project.pid}/projectFeatureFlags",
+            featureFlag: {
+              key: 'enableDataDistribution',
```
Comment (juras-gdc) [inline lib/gooddata/models/process.rb]: enableDataDistribution replaces enableEtlComponent, or the second one is discontinued ?

Comment (vu-pham) [reply lib/gooddata/models/process.rb]: it is another one, no override each other.

Comment (juras-gdc) [reply lib/gooddata/models/process.rb]: confused even more:
- before change, the elt-component wae enabled with every "deploy component" call
- now, only "data distribution" will be enabled, if and only if we process ADDv2 processes, in other words "etl component" will be never enabled, ... is this right ?

what is the purpose of these flags / what is the server behaviour when enabled  ?

Comment (vu-pham) [reply lib/gooddata/models/process.rb]: Currently, the FF enableEtlComponent and enableDataDistribution are true as default. So we can remove it, no need to set FF enableDataDistribution as well.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1553/ : SUCCESS in 10s

**Comment by vu-pham** [comment]
@juras-gdc can we merge this PR ?

**Comment by juras-gdc** [comment]
Not sure, is the test cleanup part implemented?

**Comment by juras-gdc** [comment]
Regarding slack-room, information from Danh are:
-  Jan Zdrahal told me the ruby automation test was covered for ADDv2, it’s just missing DELETE API to clean up data after test. After releasing, QA can help to verify it.

So it might be possible to merge it, if this will not break / overload mongo with dead processes, or is there some cleanup mechanism on msf side, if so, we can merge it without the cleanup part.

**Comment by vu-pham** [comment]
@juras-gdc if you are talking about the ticket https://jira.intgdc.com/browse/TMA-1487, I already delete datasource after testing

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1555/ : SUCCESS in 26s

**Comment by juras-gdc** [comment]
ready to merge, wait for https://github.com/gooddata/gooddata-ruby/pull/1533

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1558/ : SUCCESS in 19s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/333/ : FAILURE in 20s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/334/ : SUCCESS in 20s

**Pull request merged**